datumaro/components/dataset.py

```diff
@@ -713,10 +713,29 @@ def remove(self, id: str, subset: Optional[str] = None) -> None:
 
     def filter(self, expr: str, filter_annotations: bool = False,
             remove_empty: bool = False) -> Dataset:
+        """
+        Filters-out some dataset items or annotations, using a custom filter
+        expression.
+
+        Results are stored in-place. Modifications are applied lazily.
+
+        Args:
+            expr (str) - XPath-formatted filter expression
+                (e.g. "/item[subset = 'train']",
+                "/item/annotation[label = 'cat']")
+            filter_annotations (bool) - Indicates if the filter should be
+                applied to items or annotations
+            remove_empty (bool) - When filtering annotations, allows to
+                exclude empty items from the resulting dataset
+
+        Returns: a lazy-computible wrapper around the input dataset
```

Comment:
This seems to be false, as the function delegates to `self.transform`, which always returns `self`.

Reply:
Fixed description